core/src/apps/solana/types.py

```diff
@@ -0,0 +1,75 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from enum import IntEnum
+    from typing import Any, Callable, Generic, TypeVar
+
+    from trezor.utils import BufferReader
+
+    from .transaction import Instruction
+
+    Address = tuple[bytes, int]
```

Comment:
Thanks to switching to `AddressType` I noticed a bug with showing unknown instructions - we would incorrectly show accounts from lookup tables. I will fix this bug along with the type change in about an hour or so.

Reply:
Fixed in https://github.com/trezor/trezor-firmware/pull/3359/commits/e996e0de4d0b91f2fa6f17d978f915491b6122a4.

I also noticed that unknown instruction data wasn't being displayed properly because the underlying type was `memoryview` instead of `bytes` so I changed the instruction data's type to be explicitly `memoryview` in https://github.com/trezor/trezor-firmware/pull/3359/commits/c06b13c6821ed8f01ffb94d88783cf2b4e802fed. Apparently `memoryview` can be assigned to `bytes` without pyright complaining but the `confirm_properties` call works with `bytes` only.